### ISSUE TITLE
Default values of trackfield show as empty

### ISSUE THREAD

**Issue by jvangestel** [GemsTracker/gemstracker-library#392]
If you add a default value, the value is not represented in the create form.
For instance, if you have a select field with a default of the third item, on track creation the trackfield is still '-'. If you save the track, the value will be changed to the default value.
Also, when a select field is required AND a default is set, should '-' be an option at all? (I can understand that keeping it at '-' with no default set, you can force people to think about a specific selection.

Possible fixes:
Gems\Tracker\Model\AddFieldsTransformer:r109 could ask for getEmptyFieldsData from the FieldDefinition, which in turn would look like this:
```PHP
public function getEmptyFieldsData()
{
    $fields = [];
    foreach ($this->_trackFields as $key => $field) {
        $fields[$key] = $field['gtf_field_default'];
    }
    return $fields;
}
```

While in the Gems\Tracker\Field\SelectField the empty could be removed with:
```PHP
$empty = [];
if (!$this->_fieldDefinition['gtf_required'] || $this->_fieldDefinition['gtf_field_default'] === null) {
    $empty = $this->util->getTranslated()->getEmptyDropdownArray();
}
```

Question for the team: 
- Any objections in making the default trackfield option visible on create?
- Is the removal of the '-' option expected behaviour when required is true and a default is set?

**Comment by roelfaber** [comment]
28-2-2019 meeting: No objections, agreed with proposal

**Comment by mennodekker** [comment]
@jvangestel can you commit early next week so it can be included in the upcoming 1.8.6 release?
